AsyncDisplayKit/Details/Transactions/_ASAsyncTransaction.mm

```diff
@@ -327,15 +327,15 @@ - (instancetype)initWithCallbackQueue:(dispatch_queue_t)callbackQueue
     _callbackQueue = callbackQueue;
     _completionBlock = [completionBlock copy];
 
-    _state = ASAsyncTransactionStateOpen;
+    __atomic_store_n(&_state, ASAsyncTransactionStateOpen, __ATOMIC_SEQ_CST);
   }
   return self;
 }
 
 - (void)dealloc
 {
   // Uncommitted transactions break our guarantees about releasing completion blocks on callbackQueue.
-  ASDisplayNodeAssert(_state != ASAsyncTransactionStateOpen, @"Uncommitted ASAsyncTransactions are not allowed");
+  ASDisplayNodeAssert(__atomic_load_n(&_state, __ATOMIC_SEQ_CST) != ASAsyncTransactionStateOpen, @"Uncommitted ASAsyncTransactions are not allowed");
```

Comment:
@johnepinterest seems like a macro could be good to simplify accessing this variable.  This is fine for now though.